### ISSUE TITLE
CppCheck result is not shown in Sonar

### ISSUE THREAD

**Issue by alustrement** [SonarOpenCommunity/sonar-cxx-msbuild-tasks#5]
After a successfull analysis, cppcheck-result-0.xml contains this :

```
<results>
    <error file="C:\Sandbox\NativeCalc\NativeCalc\Operations.cpp" line="22" id="zerodiv" severity="error" msg="Division by zero."/>
    <error file="C:\Sandbox\NativeCalc\NativeCalc\NativeCalc.cpp" line="12" id="variableScope" severity="style" msg="The scope of the variable &apos;a&apos; can be reduced."/>
    <error file="C:\Sandbox\NativeCalc\NativeCalc\NativeCalc.cpp" line="12" id="variableScope" severity="style" msg="The scope of the variable &apos;b&apos; can be reduced."/>
    <error file="C:\Sandbox\NativeCalc\NativeCalc\NativeCalc.cpp" line="12" id="variableScope" severity="style" msg="The scope of the variable &apos;result&apos; can be reduced."/>
    <error id="missingIncludeSystem" severity="style" msg="Cppcheck cannot find all the include files (use --check-config for details)"/>
</results>
```

but no issue is raised in SonarQube. I confirm the "Division by zero" rule is activated.

**Comment by jmecosta** [comment]
dump the analysis log to file and search for the cppcheck sensor and what report string is it take. 

also what is your sonar version?

**Comment by alustrement** [comment]
I use SonarQube 5.2 (fresh sandbox install yesterday with all latest versions)
This is all I found in the dump :

```
Task "CppCheckTask"
  CppCheckCopCommand: C:\Program Files (x86)\Cppcheck\cppcheck.exe --xml --inline-suppr --enable=all -j 8 -D__cplusplus C:\Sandbox\NativeCalc
  Total Violations: 8
  CppCheck End: 59 ms
```

**Comment by jmecosta** [comment]
exactly cxx plugin does not support 5.2, your best alternative is to try to
compile this
https://github.com/SonarOpenCommunity/sonar-cxx/pull/681

ive tested and it works. however we might not even release it to 5.2
because this means braking people statistics. you can read the details in
th pr

**Comment by alustrement** [comment]
Can you send me the jar file ? I don't have any java dev env and this will me do saving time. Thank you very much

**Comment by jmecosta** [comment]
Sure

**Comment by jmecosta** [comment]
https://github.com/SonarOpenCommunity/sonar-cxx-msbuild-tasks/releases/download/1.0-RC2/sonar-cxx-plugin-0.9.5-SNAPSHOT.jar

**Comment by alustrement** [comment]
no results in Sonar 5.2 or 5.1.2 with this jar.
no results in Sonar 5.1.2 with the 0.9.4 jar.
I am doing sth wrong ?

**Comment by jmecosta** [comment]
In 5.2 I sent you should take a look at the log and see what reports is he
trying to pick up, likely you have those defined in the property file and
its not picking up correctly the path. Could you send me the last part
after the end phase?

**Comment by alustrement** [comment]
In the generated sonar-project.properties the path is ok (verified by copy/paste) :

```
sonar.cxx.cppcheck.reportPath=C:/Sandbox/NativeCalc/.cxxresults/reports-cppcheck/*.xml
```

but you're right there is sth wrong this it :

```
09:46:58.697 INFO  - Sensor CxxCppCheckSensor
09:46:58.697 DEBUG - Using pattern 'C:\Sandbox\NativeCalc\.cxxresults\reports-cppcheck\*.xml' to find reports
09:46:58.697 DEBUG - Search string transformed from 'C:\Sandbox\NativeCalc\.cxxresults\reports-cppcheck\*.xml'
09:46:58.697 DEBUG - To ':\Sandbox\NativeCalc\.cxxresults\reports-cppcheck\*.xml' to find reports, using base Path 'C:\Sandbox\NativeCalc\NativeCalc'
09:46:58.698 WARN  - Cannot find a report for 'sonar.cxx.cppcheck.reportPath=:\Sandbox\NativeCalc\.cxxresults\reports-cppcheck\*.xml'
```

**Comment by jmecosta** [comment]
ok, i will fix it. for now you can just remove the entry
sonar.cxx.cppcheck.reportPath=C:/Sandbox/NativeCalc/.
cxxresults/reports-cppcheck/*.xml from the properties file.

**Comment by alustrement** [comment]
This works :

```
sonar.cxx.cppcheck.reportPath=C:/Sandbox/NativeCalc/.cxxresults/reports-cppcheck/cppcheck-result-0.xml
```

**Comment by jmecosta** [comment]
interesting, thought i add cover the *.xml. anyway. if you have it defined in you sonar-project.properties you can just leave it out. it will recreate it for you automatically

**Comment by jmecosta** [comment]
i can reproduce this behaviour with the 5.2 version of the cxx plugin i set.

14:45:56.915 DEBUG - Using pattern 'D:\prod\structures\Core\Common.cxxresults\reports-cppcheck*.xml' to find reports
14:45:56.915 DEBUG - Search string transformed from 'D:\prod\structures\Core\Common.cxxresults\reports-cppcheck*.xml'
14:45:56.915 DEBUG - To '.cxxresults\reports-cppcheck*.xml' to find reports, using base Path 'D:\prod\structures\Core\Common'
14:45:56.916 DEBUG - Report '.cxxresults\reports-cppcheck\cppcheck-result-0.xml'
14:45:56.916 INFO  - Processing report 'D:\prod\structures\Core\Common.cxxresults\reports-cppcheck\cppcheck-result-0.xml'
14:45:56.916 INFO  - Parsing 'Cppcheck V2' format
14:45:56.916 INFO  - Parsing 'Cppckeck V1' format

was this error in with version

**Comment by jmecosta** [comment]
same issue as here:https://github.com/SonarOpenCommunity/sonar-cxx/pull/683#issuecomment-160343526

**Comment by jmecosta** [comment]
@alustrement the now the fix is merged in the sonar-cxx plugin. it should now work properly

https://ci.appveyor.com/api/buildjobs/rncf9yqkbanbcwi5/artifacts/sonar-cxx-plugin/target/sonar-cxx-plugin-0.9.5-SNAPSHOT.jar

i will close this. please re-open if still a issue. download last version from: https://github.com/SonarOpenCommunity/sonar-cxx-msbuild-tasks/releases/tag/1.2.1-RC1
